src/main/java/org/datadog/jenkins/plugins/datadog/DatadogBuildListener.java

```diff
@@ -134,7 +133,14 @@ public final void onStarted(final Run run, final TaskListener listener) {
 
       BuildStartedEventImpl evt = new BuildStartedEventImpl(builddata, tags);
       DatadogHttpRequests.sendEvent(evt);
-      gauge("jenkins.job.waiting", builddata, "waiting", extraTags);
+
+      // Send waiting metric if item.getInQueueSince() is available
+      try {
+        builddata.put("waiting", (System.currentTimeMillis() - item.getInQueueSince()) / DatadogBuildListener.THOUSAND_LONG);
```

Comment:
Man, the docs are pretty poor.... Do we know if the NPE comes from `item` being `NULL` (I would presume so) or is it being thrown by `getInQueueSince()`?

Reply:
I'm not totally sure. A stack trace is not super helpful either.
```
WARNING hudson.model.listeners.RunListener report 
RunListener failed 
java.lang.NullPointerException 
at org.datadog.jenkins.plugins.datadog.DatadogBuildListener.onStarted(DatadogBuildListener.java:113) 
at hudson.model.listeners.RunListener.fireStarted(RunListener.java:240) 
at hudson.maven.MavenBuild$ProxyImpl2.start(MavenBuild.java:609) 
at sun.reflect.GeneratedMethodAccessor981.invoke(Unknown Source) 
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) 
at java.lang.reflect.Method.invoke(Method.java:498) 
at hudson.model.Executor$2.call(Executor.java:912) 
at hudson.util.InterceptingProxy$1.invoke(InterceptingProxy.java:23) 
at com.sun.proxy.$Proxy89.start(Unknown Source) 
at sun.reflect.GeneratedMethodAccessor981.invoke(Unknown Source) 
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) 
at java.lang.reflect.Method.invoke(Method.java:498) 
at hudson.remoting.RemoteInvocationHandler$RPCRequest.perform(RemoteInvocationHandler.java:929) 
at hudson.remoting.RemoteInvocationHandler$RPCRequest.call(RemoteInvocationHandler.java:903) 
at hudson.remoting.RemoteInvocationHandler$RPCRequest.call(RemoteInvocationHandler.java:855) 
at hudson.remoting.UserRequest.perform(UserRequest.java:212) 
at hudson.remoting.UserRequest.perform(UserRequest.java:54) 
at hudson.remoting.Request$2.run(Request.java:369) 
at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:72) 
at org.jenkinsci.remoting.CallableDecorator.call(CallableDecorator.java:19) 
at hudson.remoting.CallableDecoratorList$1.call(CallableDecoratorList.java:21) 
at jenkins.util.ContextResettingExecutorService$2.call(ContextResettingExecutorService.java:46) 
at jenkins.security.ImpersonatingExecutorService$2.call(ImpersonatingExecutorService.java:71) 
at java.util.concurrent.FutureTask.run(FutureTask.java:266) 
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) 
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) 
at java.lang.Thread.run(Thread.java:748)
```